### PR TITLE
Simulated stats revamp

### PR DESCRIPTION
**Pull request by y2kappa**
### Checklist

- [] if I'm working with the latest farms.so version, I have modified the so with the latest one from master
- [] if I'm working with the latest kamino_lending.so version, I have modified the so with the latest one from master

### CODE REVIEW

File: src/classes/obligation.ts [hunk 1]
```diff
@@ -452,12 +453,12 @@ export class KaminoObligation {
...
     }
 
-    if (!reserve.state.config.elevationGroups.includes(this.state.elevationGroup)) {
+    if (!reserve.state.config.elevationGroups.includes(elevationGroup)) {
       throw new Error(
         `User would have to downgrade the elevation group in order to be able to deposit in this reserve`
       );
     }
-    const { maxLtv, liquidationLtv } = this.getLtvForReserve(market, reserve);
+    const { maxLtv, liquidationLtv } = KaminoObligation.getLtvForReserve(market, reserve, elevationGroup);
```
Comment (andreihrs) [inline src/classes/obligation.ts:461]: is this working the same 100%?

File: src/classes/obligation.ts [hunk 2]
```diff
@@ -500,13 +543,29 @@ export class KaminoObligation {
...
+
+    const { collateralExchangeRates, cumulativeBorrowRates } = KaminoObligation.getRatesForObligation(
+      market,
+      this.state,
+      params.slot
+    );
+
+    const elevationGroup = params.elevationGroupOverride ?? this.state.elevationGroup;
+
+    console.log('newStats before userTotalBorrowBorrowFactorAdjusted', newStats.userTotalBorrowBorrowFactorAdjusted);
```
Comment (andreihrs) [inline src/classes/obligation.ts:564]: remove logs

File: src/classes/obligation.ts [hunk 3]
```diff
@@ -487,10 +488,52 @@ export class KaminoObligation {
     };
   }
 
+  updateRawDeposit(
```
Comment (andreihrs) [inline src/classes/obligation.ts:491]: what does "raw" mean in this context, I find the naming a bit unintuitive

**Review by andreihrs — COMMENTED** [review]
CI failing

File: src/classes/obligation.ts (hunk 1 above)
Comment (y2kappa) [reply src/classes/obligation.ts]: yup, that was just an indirection

**Pull request merged**